pkg/query/Query.g4

```diff
@@ -0,0 +1,24 @@
+grammar Query ;
+expression: criterion (Whitespace 'and' Whitespace expression)? EOF;
+criterion: multivariate | univariate  ;
+multivariate: Key Whitespace MultiOp Whitespace multiValues ;
+univariate: Key Whitespace UniOp Whitespace Value ;
+MultiOp: 'in' | 'notin' ;
+multiValues: '(' manyValues? ')' ;
+manyValues: Value (ValueSeparator manyValues)? ;
+UniOp: 'eq' | 'ne' | 'gt' | 'lt' | 'ge' | 'le' | 'en' ;
+Value: STRING | NUMBER | BOOLEAN | DATETIME ;
+STRING: '\'' ('\\'. | '\'\'' | ~('\'' | '\\'))* '\'' ;
+BOOLEAN: 'true' | 'false' ;
+NUMBER : SIGN? DIGIT ('.' DIGIT)?;
+SIGN: '-' | '+' ;
+DATETIME: FOUR_DIGITS '-' TWO_DIGITS '-' TWO_DIGITS 'T' TWO_DIGITS ':' TWO_DIGITS ':' TWO_DIGITS '.' FIVE_DIGITS ('Z'|(SIGN TWO_DIGITS ':' TWO_DIGITS)) ;//2016-06-08T17:41:22Z
+DIGIT: INTEGER+ ;
+INTEGER: [0-9] ;
+TWO_DIGITS: DIGIT DIGIT ;
+FOUR_DIGITS: TWO_DIGITS TWO_DIGITS ;
+FIVE_DIGITS: FOUR_DIGITS DIGIT ;
+Key: [-_/a-zA-Z0-9\\]+ ;
+ValueSeparator: ',' | ', ' ;
+Whitespace: ' ' ;
+WS : [ \t\r\n]+ -> skip ; // skip spaces, tabs, newlines
```

Comment:
I adjusted most of the comments in the gammar. Only thing left is the date - there are two options - either something more readable such as https://github.com/antlr/grammars-v4/blame/76873507fe61c9246831d7e903cb64d19075ece4/toml/toml.g4#L95 or the regex in the comment above

```
grammar Query ;

expression: criterions EOF ;
criterions: criterion (Concat criterions)? ;
criterion: multivariate | univariate  ;
multivariate: Key  MultiOp  multiValues ;
univariate: Key UniOp Value ;
multiValues: OpenBracket manyValues? CloseBracket ;
manyValues: Value (ValueSeparator manyValues)? ;

MultiOp: Whitespace 'in' Whitespace | Whitespace 'notin' Whitespace;
UniOp: Whitespace 'eq' Whitespace| Whitespace 'ne' Whitespace | Whitespace 'gt' Whitespace | Whitespace 'lt' Whitespace | Whitespace 'ge' Whitespace | Whitespace 'le' Whitespace | Whitespace 'en' Whitespace;
Concat: Whitespace 'and' Whitespace ;
Value: STRING | NUMBER | BOOLEAN | DATETIME ;
ValueSeparator: ',' | ', ' ;
Key: [-_/a-zA-Z0-9\\]+ ;
OpenBracket: '(' | '[' ;
CloseBracket: ')' | ']' ;

fragment BOOLEAN: 'true' | 'false' ;
fragment STRING: '\'' ('\\'. | '\'\'' | ~('\'' | '\\'))* '\'' ;
fragment DATETIME: FOUR_DIGITS '-' TWO_DIGITS '-' TWO_DIGITS 'T' TWO_DIGITS ':' TWO_DIGITS ':' TWO_DIGITS '.' FIVE_DIGITS ('Z'|(SIGN TWO_DIGITS ':' TWO_DIGITS)) ;//2016-06-08T17:41:22Z
fragment FIVE_DIGITS: FOUR_DIGITS DIGIT ;
fragment FOUR_DIGITS: TWO_DIGITS TWO_DIGITS ;
fragment TWO_DIGITS: DIGIT DIGIT ;
fragment NUMBER : SIGN? DIGIT ('.' DIGIT)?;
fragment SIGN: '-' | '+' ;
fragment DIGIT: INTEGER+ ;
fragment INTEGER: [0-9] ;
fragment Whitespace: ' ' ;

WS: [ \t\r\n]+ -> skip ; // skip spaces, tabs, newlines
```

Reply:
Seems strange for the operator to include the whitespace in its definition. Would require processing of whitespaces in the parser